lib/lint/reporter/process-warns.js

```diff
@@ -16,5 +21,36 @@ module.exports = function(warns, fileFilter){
     });
   });
 
+  if (!isChildProcess)
```

Comment:
Bad place for that logic. This module process/aggregate warnings, and should not test for process type or collect a new warnings.
In case we need add new warnings, you can add them into `flow.warns` directly